### DISCUSSION TITLE
Chocolatey product stable releases (v2.0.0 and v6.0.0)

### DISCUSSION THREAD

**Discussion by gep13** [chocolatey/choco#3173] [General]
If you are running into any issues with the new release of Chocolatey products, please leave a comment in this discussion.

**Comment by mfnones** [comment]
Good day! We are having issue with the 2.0.0 version as seen on the image below. We are using azure container dockerfile to execute this, so I am not yet aware on how to do the ".NET Framework 4.8 was installed, but a reboot is required." Is there a way for me to point the install.ps1 to a 1.4.0 version, while we are looking on how to fix the issue on reboot? Thank you!
![image_2023_06_01T06_13_32_345Z](https://github.com/chocolatey/choco/assets/65269476/2ffcce48-8bb0-41be-b9f2-2c29f1e77b1c)

**Reply by gep13** [reply]
The best suggestion would be to use a base docker image that already has .NET Framework 4.8 already installed on it.  Assuming that you are referring to this [install.ps1](https://community.chocolatey.org/install.ps1) then yes, you will see that there are arguments that can be passed in, which control the version of Chocolatey that is installed.

**Reply by pauby** [reply]
I've added a section to the [upgrade guide on .NET 4.8 and Docker](http://localhost:5080/en-us/guides/upgrading-to-chocolatey-v2-v6#net-framework-4.8-required).

**Comment by sixside** [comment]
Recently upgraded, and now when I try to open GUI, I'm getting this error dialog:

"Unable to load Chocolatey GUI assembly. The type initializer for 'ChocolateyGui.Common.Windows.Bootstrapper' threw an exception."

And the GUI doesn't load. I searched for older posts on the issue and some suggested using the command line to uninstall/reinstall the GUI, which I did. But still getting the same error. Any help on a resolution would be appreciated!

**Reply by gep13** [reply]
Can you confirm exactly what packages you have installed?  Assuming that you made it to Chocolatey CLI v2.0.0, you can run:

```
choco list
```

If you are still on an older version, you can run:

```
choco list --local-only
```

Can I also get you to try running `chocolateyguicli` from a PowerShell session, and report back the complete exception that is reported, if there is an exception thrown.

**Reply by sixside** [reply]
C:\WINDOWS\system32> choco list --local-only
Chocolatey v1.3.0
... other apps ...
chocolatey 2.0.0
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
chocolatey-dotnetfx.extension 1.0.1
chocolatey-windowsupdate.extension 1.0.5
chocolateygui 2.0.0


 C:\WINDOWS\system32> chocolateyguicli
Method not found: 'chocolatey.infrastructure.licensing.ChocolateyLicense chocolatey.infrastructure.licensing.License.ValidateLicense()'.

**Reply by gep13** [reply]
@sixside based on this output, it would appear that for some reason, the Chocolatey CLI 2.0.0 has been installed along with Chocolatey GUI 2.0.0, however, Chocolatey CLI 1.3.0 is still being run.  I am not sure why this would have happened, but to get back to a known state, can I suggest you do the following...

Uninstall Chocolatey GUI
```choco uninstall chocolateygui```

Downgrade to an earlier version of Chocolatey CLI:
```choco upgrade chocolatey --version 1.4.0 --allow-downgrade```

Upgrade to the latest Chocolatey CLI:
```choco upgrade chocolatey```

Install Chocolatey GUI again:
```choco install chocolateygui```

And hopefully that will get you to a place where everything is running.

**Reply by sixside** [reply]
Brilliant! Thanks so much, greatly appreciated your helpful reply.

FWIW, I only ever upgrade Chocolatey and the GUI from within the GUI itself when prompted with a new version. So whatever broke must have been during the prior upgrade.

**Reply by Unknown-Guy** [reply]
Same here, I do not fallow Chocolatey development, the GUI popped up with upgrade notification and after that I ended up with the same error.

**Reply by gep13** [reply]
I believe that this is something that we need to improve on, as updating Chocolatey CLI from within Chocolatey GUI can/will cause problems.

@Unknown-Guy can you confirm if you have been able to get back to a working state?

**Reply by Unknown-Guy** [reply]
@gep13 all good now, the instructions you have above worked great.

**Comment by mikeboutch** [comment]
upgrade is stuck and never finish!
```

λ clist -lo |grep -i ^choco
Chocolatey v1.4.0 Business
choco-cleaner 0.0.8.4
choco-GEDconfig-dev 22.9.1
Choco-License-GED 22.9.1
chocolatey 1.4.0
chocolatey-agent 1.1.2
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
chocolatey-dotnetfx.extension 1.0.1
chocolatey-windowsupdate.extension 1.0.5
chocolatey.extension 5.0.3
chocolateygui 1.1.3
chocolateygui.extension 1.0.3
                                                                                                                    
C:\tools\Cmder                                                                                                      
λ  choco upgrade chocolatey chocolatey.extension chocolatey-agent chocolateyGUI chocolateyGUI.extension -y -f       
Chocolatey v1.4.0 Business                                                                                          
Upgrading the following packages:                                                                                   
chocolatey;chocolatey.extension;chocolatey-agent;chocolateyGUI;chocolateyGUI.extension                              
By upgrading, you accept licenses for the packages.                                                                 
                                                                                                                    
You have chocolatey v1.4.0 installed. Version 2.0.0 is available based on your source(s).      
```                     
 and    try  to wait 1 hour and nothing!                                                                                                        
[choco.log](https://github.com/chocolatey/choco/files/11807619/choco.log)

**Reply by pauby** [reply]
@mikeboutch As you're a licensed customer, can you reach out to support? Run 'choco support' to find out your options.

**Reply by mikeboutch** [reply]
I fix it by adding ---sdc skip download cache
